### PR TITLE
JAPI-135 Standardize soapproxy access within wsclient code

### PR DESCRIPTION
**Pull request by rpastrana**
- Replaces direct soapaccess null tests with function call

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java [hunk 1]
```diff
@@ -196,58 +194,55 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
...
 
-                try
+            try
+            {
+                WriteFileDataResponse writeFileDataResponse = wsFileIOServiceSoapProxy.writeFileData(writefileparams);
+                String result = writeFileDataResponse.getResult();
+
+                log.debug(result);
+                //Wish there was a better way
+                if (!result.startsWith("Failed"))
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java]: That does seem brittle. Do we have a ticket to create a better status code?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java [hunk 2]
```diff
@@ -196,58 +194,55 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
...
-                    }
+                    success = true;
+                    bytesleft = bytesleft - subdata.length;
                 }
-                catch (Exception e)
+                else
                 {
-                    log.error(e.getLocalizedMessage());
-                    return false;
+                    bytesleft = 0;
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java]: Should there be a log message or an exception thrown here?

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java [hunk 3]
```diff
@@ -196,58 +194,55 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
...
-            {
-                payloadsize = bytesleft >= limit ? limit : bytesleft;
-                log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
+            payloadsize = bytesleft >= limit ? limit : bytesleft;
+            log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
 
-                subdata = new byte [(int)payloadsize];
-                for (int i = 0; i < payloadsize ; i++, dataindex++)
-                    subdata[i] = data[dataindex];
+            subdata = new byte [(int)payloadsize];
```
[the remark this answers is not in the document]
Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java]: https://track.hpccsystems.com/browse/JAPI-144

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java (hunk 1 above)
Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java]: agreed, requested change in the ESP side: https://track.hpccsystems.com/browse/HPCC-21293
but unlikely to change.

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java (hunk 2 above)
Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java]: I think the else condition implies success.Anyhow this logic didn't change, it was indented due to removal of if/else

**Comment by rpastrana** [comment]
@Michael-Gardner please merge

**Pull request merged**